web/js/students_of_class.js

```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
+        female: formData.get("female") === "female",
+        schoolClass: formData.get("schoolClass")
     };
-    addStudent(newStudent)
+    await addStudent(data)
         .catch(() => {
             errorElement.innerHTML = "The post request was not successful.";
             $(errorElement).slideDown().delay(3000).slideUp();
-        })
-}
-
-function isFemale() {
-    const femaleOption = document.getElementById("addFemale");
-    if (femaleOption.value === "female") {
-        return true;
-    } else if (femaleOption.value === "male") {
-        return false;
-    }
-}
-
-function stillFemale() {
-    const femaleOption = document.getElementById("editFemale");
-    if (femaleOption.value === "female") {
-        return true;
-    } else if (femaleOption.value === "male") {
-        return false;
-    }
+        });
+    await fetchApi();
 }
 
 function editOrAdd() {
-    const editOption = document.getElementById("EditOrAdd");
+    const editOption = document.querySelector("#EditOrAdd");
     if (editOption.value === "edit") {
         return true;
     } else if (editOption.value === "add") {
         return false;
     }
 }
 
-function editStudent() {
-    modalEditStudent.modal('hide');
+async function editStudent(formData) {
     const errorElement = document.querySelector("#error").value;
-    const editFirstName = document.getElementById("editFirstName").value;
-    const editLastName = document.getElementById("editLastName").value;
-    const editBirthday = document.getElementById("editBirthday").value;
-    const editFemale = stillFemale();
-    const edits = {firstName: editFirstName, lastName: editLastName, birthDay: editBirthday, female: editFemale};
-    const studentURL = document.getElementById("studentURL").value;
-    patchStudent(studentURL, edits)
+    modalEditStudent.modal('hide');
+    const data = {
+        firstName: formData.get("editFirstName"),
+        lastName: formData.get("editLastName"),
+        birthDay: formData.get("editBirthDay"),
+        female: formData.get("editFemale") === "female"
+    };
+    const studentURL = formData.get("studentURL");
+    await patchStudent(studentURL, data)
         .catch(() => {
             errorElement.innerHTML = "The patch request was not successful.";
             $(errorElement).slideDown().delay(3000).slideUp();
-        })
-}
-
-function ShowHideSaveButton(promise) {
-    promise ? document.getElementById("saveSportResultButton").style.visibility = 'visible' : document.getElementById("saveSportResultButton").style.visibility = 'hidden';
+        });
+    await fetchApi();
 }
 
 function updateSchoolClass(schoolClass) {
-    const className = document.getElementById("class-name");
+    const className = document.querySelector("#class-name");
     className.innerHTML = `${schoolClass.grade}${schoolClass.className}`;
 
-    const classTitle = document.getElementById("title");
+    const classTitle = document.querySelector("#title");
```

Comment:
Please rename this to `websiteTitle` or anything similar. This is quite confusing ...